src/Func.cpp

```diff
@@ -1616,6 +1616,10 @@ Func &Func::unroll(VarOrRVar var, int factor, TailStrategy tail) {
 }
 
 Func &Func::bound(Var var, Expr min, Expr extent) {
+    user_assert(!min.defined() || (min.type() == Int(32))) << "Min bound of a Func must be of type int32\n";
```

Comment:
This looks like it is going to insist on a 32-bit integer, which has a couple issues. First with the 64bit coordinates work, we should likely support int64_t, and second it will be fairly easy to pass an 8 or 16 bit value that causes this to error out. Do we explicitly want to require that strictness or to allow IntImm and UIntImm values that can be losslessly converted to int32_t?